README.md
<img src="https://cdn.rawgit.com/brigadehub/brigadehub/develop/assets/rasterized/logo-banner-color-white.png" alt="Brigadehub"/>

---

[![TrustOSS Compliant](http://trustoss.org/badge_version.svg)](http://trustoss.org)
[![Build Status](https://travis-ci.org/brigadehub/brigadehub.svg?branch=edge)](https://travis-ci.org/brigadehub/brigadehub)
[![npm](https://img.shields.io/npm/v/brigadehub.svg?maxAge=2592000)](https://www.npmjs.com/package/brigadehub)
[![Slack Status](https://sfbrigade-slackin.herokuapp.com/badge.svg)](https://sfbrigade-slackin.herokuapp.com/)
[![Keep A Changelog](https://img.shields.io/badge/Keep%20A-Changelog-blue.svg?style=flat)](http://keepachangelog.com/)
[![JS Standard](https://img.shields.io/badge/JS-Standard-yellow.svg?style=flat)](https://github.com/feross/standard)

### Table of Contents

- [In a Nutshell](#in-a-nutshell)
- [Ways to Use](#ways-to-use)
  - [Suite](#suite)
    - [Production Install](#production-installation)
    - [Development Install](#development-installation)
  - [Admin Gateway](#admin-gateway)
    - [Production Install](#production-installation)
    - [Development Install](#development-installation)
  - [Mini](#mini)
    - [Production Install](#production-installation)
    - [Development Install](#development-installation)
- [Usage](#usage)
  - [Core API](#core-api)
  - [Admin Theme](#admin-theme)
  - [Public Theme](#public-theme)
- [Troubleshooting](#troubleshooting)
- [Contributing](#contributing)
- [Changelog](#changelog)
- [Contributors](#contributors)
- [Inspirations](#inspirations)
- [Additional Resources](#resources)
- [License](#license)

### In a nutshell

BrigadeHub is an **beta** stage data portal for volunteer coding organizations. While being built for Code for America Brigades, it can be be used for any project-based coding organization. It's being built to consolidate the efforts of maintaining a brigade's information into a single location, to distribute the task of content creation across the various non-technical brigade roles, and to enhance the experience of both members and admins of these unique organizations.

For a complete top-down view of the roadmap, take a look at our active ***[Roadmap Wiki](https://github.com/brigadehub/brigadehub/wiki/Roadmap)*** (currently being worked on).

In short, BrigadeHub is designed to meet some very specific goals:

- Provide out-of-the-box utilities for organization admins:
  - Public facing website, with contact page, blog, project profiles and more
  - Admin dashboard
  - Robust content management permissioning and distribution
- Be easy to install and upkeep. You shouldn't have to dedicate a team just for your website anymore, nor should you need to know how to code to update your site.
- Be modular, extensible, and hackable. Brigadehub Core provides a simple REST API for minimal deployment if you already have your infrastructure in place.
- Allow brigades to communicate with each other quickly and effectively

### Ways to use

There are three different ways of interacting with/using Brigadehub:

- **Suite**: out of the box solution, best if you have no current site solution and don't want to deal with any coding. Swap out themes as you like with community-built solutions.
- **Admin Gateway**: Minimal API with a GUI admin gateway, best if you already have a website and want a way of dynamically generating the content of it. You'll need to read up on the Core API to tie it into your frontend, but once deployed, the data can be maintained by non-coders.
- **Mini**: This is just a REST API interface. Best if you already have an interface built out and just want to leverage the dataset of Brigadehub.

Depending on the needs of your organization, you'll want to decide which will work best for you.

#### Suite

**Brigadehub Suite** is the flagship install of Brigadehub, and is more or less a ***brigade-in-a-box***. It installs three components: **Core**, **Admin Theme**, and **Public Theme**. It's deployed in such a way that you should never *need* to touch the code, but if you *want* to, it's a simple matter. It provides Admin tools to manage your brigade information, and a slick frontend to display the info you're managing. Your frontend developers can dynamically pull the data about your brigade via the Core API if they have external tools that need the same data. And you can rest assured that we'll be working on patches and updates over time to make this the most secure and stable platform ever.

###### Production Install

<details><summary><img src="http://saasiter.com/img/services/heroku.png" height="25" /> Heroku (click to open)</summary><p>
Heroku sports this handy one-click-install feature, which will handle most of the configuration for you. Base the install off [brigadehub/brigadehub](https://github.com/brigadehub/brigadehub), and you're good to go (the button below does that ⬇)

[![Brigadehub Suite Heroku Install](https://www.herokucdn.com/deploy/button.svg)](https://dashboard.heroku.com/new?template=https%3A%2F%2Fgithub.com%2Fbrigadehub%2Fbrigadehub%2Ftree%2Fmaster)

</p></details><br/>

<details><summary><img src="https://pbs.twimg.com/profile_images/378800000124779041/fbbb494a7eef5f9278c6967b6072ca3e_normal.png" height="25" /> Docker (click to open)</summary><p>
***Pre-requisites***

- [Docker Engine](https://docs.docker.com/engine/installation/)
- A running mongo instance, via docker or otherwise

***Installation and Running***

```bash
docker pull brigadehub/suite
docker run -d --expose=8080:desiredport -e MONGODB='mongodb://location.of.mongo:27017/brigadehub-suite' -e PORT=desiredport brigadehub/suite
```

</p></details><br/>

<details><summary><img src="https://www.npmjs.com/static/images/touch-icons/favicon-32x32.png" height="25" /> NPM CLI (click to open)</summary><p>
***Pre-requisites***

- [Node 6+](https://nodejs.org/en/)
- A running mongo instance

***Installation and Running***
```bash
npm install -g brigadehub
PORT=desiredport MONGODB_URI=mongodb://location.of.mongo:27017/brigadehub-suite brigadehub-suite
```
</p></details><br/>

<details><summary><img src="http://lepbase.org/wp-content/themes/lepbase-dot-org/images/code-icon.png" height="25" /> Source (click to open)</summary><p>
***Pre-requisites***

- Git
- [Node 6+](https://nodejs.org/en/)
- A running mongo instance

***Installation and Running***
```bash
git clone https://github.com/brigadehub/brigadehub
cd brigadehub
make install
PORT=desiredport MONGODB_URI=mongodb://location.of.mongo:27017/brigadehub-suite make start
```
</p></details>

###### Development Install

<details><summary><img src="http://lepbase.org/wp-content/themes/lepbase-dot-org/images/code-icon.png" height="25" /> Source (click to open)</summary><p>
To work on Brigadehub Suite as a developer, you need to clone and link internally all four portions of the suite. Fork all 4 repos into your desired organization or account, and work from those, making sure to update from upstream periodically.

***Pre-requisites***

- Git
- [Node 6+](https://nodejs.org/en/)
- A running mongo instance

***Installation and Running***

```bash
git clone https://github.com/brigadehub/brigadehub
git clone https://github.com/brigadehub/core
git clone https://github.com/brigadehub/theme-public-c4sf
git clone https://github.com/brigadehub/theme-admin-c4sf
```

In separate terminals, run the following:

```bash
cd theme-public-c4sf
make install
make link
```

```bash
cd theme-admin-c4sf
make install
make link
```

```bash
cd core
make install
make link
```

```bash
cd brigadehub
make install
make link
MONGODB_URI=mongodb://location.of.mongo:27017/brigadehub-suite make start/develop
```

Any changes made in core, theme-public-c4sf, and theme-admin-c4sf will restart the server in brigadehub, and allow you to preview the changes. When you're done with your changes, make sure the tests pass via `make test`, and create a pull request from the appropriate repo.

</p></details>

#### Admin Gateway

**Brigadehub Admin Gateway** installs two components: **Core** and **Admin Theme**. Your frontend developers can dynamically pull the data about your brigade via the Core API, but you as an administrator should never need to touch the code. This is a good option if you already have a site and want to augment it with brigadehub data.

###### Production Install

<details><summary><img src="http://saasiter.com/img/services/heroku.png" height="25" /> Heroku (click to open)</summary><p>
Heroku sports this handy one-click-install feature, which will handle most of the configuration for you. Base the install off [brigadehub/admin-gateway](https://github.com/brigadehub/admin-gateway), and you're good to go (the button below does that ⬇)

[![Brigadehub Admin Gateway Heroku Install](https://www.herokucdn.com/deploy/button.svg)](https://dashboard.heroku.com/new?template=https%3A%2F%2Fgithub.com%2Fbrigadehub%2Fadmin-gateway%2Ftree%2Fmaster)
</p></details><br/>

<details><summary><img src="https://pbs.twimg.com/profile_images/378800000124779041/fbbb494a7eef5f9278c6967b6072ca3e_normal.png" height="25" /> Docker (click to open)</summary><p>
***Pre-requisites***

- [Docker Engine](https://docs.docker.com/engine/installation/)
- A running mongo instance, via docker or otherwise

***Installation and Running***

```bash
docker pull brigadehub/admin-gateway
docker run -d --expose=8080:desiredport -e MONGODB='mongodb://location.of.mongo:27017/brigadehub-admin-gateway' -e PORT=desiredport brigadehub/admin-gateway
```

</p></details><br/>

<details><summary><img src="https://www.npmjs.com/static/images/touch-icons/favicon-32x32.png" height="25" /> NPM CLI (click to open)</summary><p>
***Pre-requisites***

- [Node 6+](https://nodejs.org/en/)
- A running mongo instance

***Installation and Running***
```bash
npm install -g brigadehub-admin-gateway
brigadehub-admin-gateway --port desiredport --mongodb mongodb://location.of.mongo:27017/brigadehub-admin-gateway
```
</p></details><br/>

<details><summary><img src="http://lepbase.org/wp-content/themes/lepbase-dot-org/images/code-icon.png" height="25" /> Source (click to open)</summary><p>
***Pre-requisites***

- Git
- [Node 6+](https://nodejs.org/en/)
- A running mongo instance

***Installation and Running***
```bash
git clone https://github.com/brigadehub/admin-gateway
cd admin-gateway
make install
PORT=desiredport MONGODB_URI=mongodb://location.of.mongo:27017/brigadehub-admin-gateway make start
```
</p></details>

###### Development Install

<details><summary><img src="http://lepbase.org/wp-content/themes/lepbase-dot-org/images/code-icon.png" height="25" /> Source (click to open)</summary><p>
To work on Brigadehub Admin Gateway as a developer, you need to clone and link internally all three portions of the install. Fork all 3 repos into your desired organization or account, and work from those, making sure to update from upstream periodically.

***Pre-requisites***

- Git
- [Node 6+](https://nodejs.org/en/)
- A running mongo instance

***Installation and Running***

```bash
git clone https://github.com/brigadehub/admin-gateway
git clone https://github.com/brigadehub/core
git clone https://github.com/brigadehub/theme-admin-c4sf
```

In separate terminals, run the following:

```bash
cd theme-admin-c4sf
make install
make link
```

```bash
cd core
make install
make link
```

```bash
cd admin-gateway
make install
make link
MONGODB_URI=mongodb://location.of.mongo:27017/brigadehub-admin-gateway make start/develop
```

Any changes made in core and theme-admin-c4sf will restart the server in admin-gateway, and allow you to preview the changes. When you're done with your changes, make sure the tests pass via `make test`, and create a pull request from the appropriate repo.

</p></details>

#### Mini

**Brigadehub Mini** installs only one component: **Core**. No Frontends will be deployed with this install, exposing only the authentication routes and REST APIs. This is an advanced setting, and should be only done if you already have a site and admin portal to manage on your own. All features are accessible via REST API (GraphQL to come soon)

###### Production Install

<details><summary><img src="http://saasiter.com/img/services/heroku.png" height="25" /> Heroku (click to open)</summary><p>
Heroku sports this handy one-click-install feature, which will handle most of the configuration for you. Base the install off [brigadehub/mini](https://github.com/brigadehub/mini), and you're good to go (the button below does that ⬇)

[![Brigadehub Admin Gateway Heroku Install](https://www.herokucdn.com/deploy/button.svg)](https://dashboard.heroku.com/new?template=https%3A%2F%2Fgithub.com%2Fbrigadehub%2Fmini%2Ftree%2Fmaster)
</p></details><br/>

<details><summary><img src="https://pbs.twimg.com/profile_images/378800000124779041/fbbb494a7eef5f9278c6967b6072ca3e_normal.png" height="25" /> Docker (click to open)</summary><p>
***Pre-requisites***

- [Docker Engine](https://docs.docker.com/engine/installation/)
- A running mongo instance, via docker or otherwise

***Installation and Running***

```bash
docker pull brigadehub/mini
docker run -d --expose=8080:desiredport -e MONGODB='mongodb://location.of.mongo:27017/brigadehub-mini' -e PORT=desiredport brigadehub/mini
```
</p></details><br/>

<details><summary><img src="https://www.npmjs.com/static/images/touch-icons/favicon-32x32.png" height="25" /> NPM CLI (click to open)</summary><p>
***Pre-requisites***

- [Node 6+](https://nodejs.org/en/)
- A running mongo instance

***Installation and Running***
```bash
npm install -g brigadehub-mini
brigadehub-mini --port desiredport --mongodb mongodb://location.of.mongo:27017/brigadehub-mini
```
</p></details><br/>

<details><summary><img src="http://lepbase.org/wp-content/themes/lepbase-dot-org/images/code-icon.png" height="25" /> Source (click to open)</summary><p>
***Pre-requisites***

- Git
- [Node 6+](https://nodejs.org/en/)
- A running mongo instance

***Installation and Running***
```bash
git clone https://github.com/brigadehub/mini
cd mini
make install
PORT=desiredport MONGODB_URI=mongodb://location.of.mongo:27017/brigadehub-mini make start
```
</p></details>

###### Development Install

<details><summary><img src="http://lepbase.org/wp-content/themes/lepbase-dot-org/images/code-icon.png" height="25" /> Source (click to open)</summary><p>
To work on Brigadehub Mini as a developer, you need to clone and link internally both portions of the install. Fork both repos into your desired organization or account, and work from those, making sure to update from upstream periodically.

***Pre-requisites***

- Git
- [Node 6+](https://nodejs.org/en/)
- A running mongo instance

***Installation and Running***

```bash
git clone https://github.com/brigadehub/mini
git clone https://github.com/brigadehub/theme-admin-c4sf
```

In separate terminals, run the following:

```bash
cd core
make install
make link
```

```bash
cd mini
make install
make link
MONGODB_URI=mongodb://location.of.mongo:27017/brigadehub-mini make start/develop
```

Any changes made in core and theme-admin-c4sf will restart the server in mini, and allow you to preview the changes. When you're done with your changes, make sure the tests pass via `make test`, and create a pull request from the appropriate repo.

</p></details>

#### [Usage](#usage)
##### [Core API](#core-api)
*Core API Documentation is being worked on*
##### [Admin Theme](#admin-theme)
*Admin Theme Documentation is being worked on*
##### [Public Theme](#public-theme)
*Public Theme Documentation is being worked on*
#### [Troubleshooting](#troubleshooting)
*If you come upon issues while installing or using Brigadehub (any install), open an issue here in Github. If it occurs enough times, it'll make it here into Troubleshooting*
#### [Contributing](#contributing)

Brigadehub works from a fork/PR methodology. If you would like to contribute, check the github issues and see if someone is working on the issue already. If not, comment in the issue that you are working on it (or create a new issue if it's not already there), then fork the repo and submit a PR when your work is complete. Claimed issues without update on the issue itself or via linked PR after 30 days will be unassigned and reopened for others to work on.

#### [Changelog](#changelog)
The changelog for Brigadehub (and each of its components) resides in the appropriate repos ***Releases*** tab.

#### [Contributors](#contributors)

Brigadehub is a Code for San Francisco Infrastructure project, and has been brought about by the generous work of the following individuals. If you make a contribution to Brigadehub (e.g. when opening a PR), feel free to add yourself to this list in addition to your changes. If you are contributing in a non-code fashion, open an issue to have your name added.

- Oz Haven - Project Lead - ([@therebelrobot](https://github.com/therebelrobot)) <`github``@``therebelrobot.com`>
- Todd Seller - Project Co-Lead - [@toddseller](https://github.com/toddseller)
- Aaron Schachter [@aaronschachter](https://github.com/aaronschachter)
- Andrew Bacon [@andrewmbacon](https://github.com/andrewmbacon)
- David Carvel [@davidcarvel](https://github.com/davidcarvel)
- David Hincapie [@davidhincapie](https://github.com/davidhincapie)
- Deborah [@deborahahn](https://github.com/deborahahn)
- Debbie Espino [@despino](https://github.com/despino)
- Gisela [@GiselaKay](https://github.com/GiselaKay)
- Leo Lau [@hawflau](https://github.com/hawflau)
- Jason Durant [@jaydurant](https://github.com/jaydurant)
- John Naulty Jr. [@jnaulty](https://github.com/jnaulty)
- John Gluck [@johngluck65](https://github.com/johngluck65)
- Jesse Szwedko [@jszwedko Owner](https://github.com/jszwedko Owner)
- Jessica Parsons [@verythorough](https://github.com/verythorough)
- Judy5000 [@Judy5000 Owner](https://github.com/Judy5000 Owner)
- Justin [@Juxtaposeidon](https://github.com/Juxtaposeidon)
- kanagha [@kanagha](https://github.com/kanagha)
- Kevin Litchfield [@kevinlitchfield](https://github.com/kevinlitchfield)
- Maja Bogeski [@majabogeski](https://github.com/majabogeski)
- Neri J. Jakubowski Junior [@nerijunior](https://github.com/nerijunior)
- Patrick Connolly [@patcon](https://github.com/patcon)
- Phips Peter [@pspeter3](https://github.com/pspeter3)
- [@rogeliobonilla](https://github.com/rogeliobonilla)
- Tracy [@spiffysparrow](https://github.com/spiffysparrow)
- [@Thithi32](https://github.com/Thithi32)
- [@whatdoublechen](https://github.com/whatdoublechen)
- [@wrendo](https://github.com/wrendo)

#### Inspirations

Similar projects have been conceived and implemented previously ([Laddr](https://github.com/CfABrigadePhiladelphia/laddr), [Connector](https://github.com/codeforatlanta/connector), [Chime](https://github.com/chimecms/chime ), etc.), but for one reason or another didn't meet the needs of Code for San Francisco and many other brigades. Brigadehub hopes to bridge those infrastructure gaps.

#### [Additional Resources](#resources)

Have additional resources that could benefit those using Brigadehub? Open a Github Issue and let us know!

#### [License](#license)

[MIT](https://tldrlegal.com/license/mit-license)

The MIT License (MIT)

Copyright (c) 2016-2017 Code for San Francisco

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.


.github/CHANGELOG.md

# v1.0.0-alpha.19 - Vicious Scorpion | published *2016-12-07T16:55:08Z*

## 2016-12-07, Version v1.0.0-alpha.19
### Notable Changes

- Enter Notable Changes Here

### Commits
- [[`ae7581f33d`](https://github.com/brigadehub/brigadehub/commit/ae7581f33da19a10898ebdef4c121fce5babb89f)] - **misc:**  various updates and fixes (Trent Oswald)
- [[`e1c2668d78`](https://github.com/brigadehub/brigadehub/commit/e1c2668d78423a7ed597bc852b482d57ee289576)] - **version:**  bumped to 1.0.0-alpha.18 (Trent Oswald)
- [[`898c7cc2dd`](https://github.com/brigadehub/brigadehub/commit/898c7cc2dde6a5268121ca20f6231778b924602c)] - **version:**  bumped to 1.0.0-alpha.19 (Trent Oswald)
- [[`bd551842d8`](https://github.com/brigadehub/brigadehub/commit/bd551842d8145dd6ac880beeb74b9e403ed8a313)] - **lint:**  standard compliance (Trent Oswald)


# v1.0.0-alpha.18 - Sharp Fish | published *2016-12-06T21:51:23Z*

## 2016-12-06, Version v1.0.0-alpha.18
### Notable Changes

- Stability and theme version updates

### Commits
- [[`432bd0760c`](https://github.com/brigadehub/brigadehub/commit/432bd0760ccfdfdd98637205236eea0a9da9340a)] - **theme:**  updated public theme version (Trent Oswald)
- [[`521e94816c`](https://github.com/brigadehub/brigadehub/commit/521e94816c14111f2fd3ceb1b60996fe07f3bf44)] - **blog:**  various styling and model fixes to blog (Trent Oswald)
- [[`5914169aaa`](https://github.com/brigadehub/brigadehub/commit/5914169aaa7ba0c131527bf9d25ec6a035435c2f)] - **lint:**  standard compliance (Trent Oswald)
- [[`f30aec54a6`](https://github.com/brigadehub/brigadehub/commit/f30aec54a61f065b283fe3495d9795d72f421b00)] - **theme:**  updated theme version (Trent Oswald)
- [[`321f5c1805`](https://github.com/brigadehub/brigadehub/commit/321f5c180521b3bd9836b54fcbea3a1c67a98f23)] - **theme:**  Update public theme version (Todd Seller)
- [[`7de3a378e3`](https://github.com/brigadehub/brigadehub/commit/7de3a378e3bae3f4e1bf2c10730c53d2b9e439b4)] - **blog:**  fixed blog issues (Trent Oswald)
- [[`87114e55a4`](https://github.com/brigadehub/brigadehub/commit/87114e55a402c24a9f86dcafbb5df455c4d0bf4b)] - **deps:**  updated public theme version (Trent Oswald)
- [[`0cd6bcb00f`](https://github.com/brigadehub/brigadehub/commit/0cd6bcb00fc1fd79bd1f58b56266247496ca5b67)] - **lint:**  standard compliance (Trent Oswald)
- [[`2ffd32a8a8`](https://github.com/brigadehub/brigadehub/commit/2ffd32a8a899c87477faa7617e3d31e79286c0a0)] - **theme-update:**  Updates public theme (Todd Seller)
- [[`4cc1416824`](https://github.com/brigadehub/brigadehub/commit/4cc14168244367c4c4269259d89e29e4de722f21)] - **projects:**  fixed pre-save finction for project members (Trent Oswald)


# v1.0.0-alpha.17 - Beautiful Cheetah | published *2016-12-06T22:02:00Z*

## 2016-12-06, Version v1.0.0-alpha.17
### Notable Changes

- Model fixes

### Commits
- [[`158d938357`](https://github.com/brigadehub/brigadehub/commit/158d9383571d78cd89100a88286878b6df8b0465)] - **deps:**  making sure db-migrate is in normal deps (Trent Oswald)
- [[`8fd2f8e432`](https://github.com/brigadehub/brigadehub/commit/8fd2f8e4328cf1641d846f2b5ace299288f7e625)] - **start:**  removed gulp build from start script (Trent Oswald)
- [[`1941653db2`](https://github.com/brigadehub/brigadehub/commit/1941653db2291844a51f56cde4f824d8969c14cb)] - **bootstrap:**  fixing brigade model bootstrapping (Trent Oswald)
- [[`3f3739fb59`](https://github.com/brigadehub/brigadehub/commit/3f3739fb59bdcc90d67b1266c78129df46045d20)] - **model:**  updating model references (Trent Oswald)
- [[`e7f12a7acf`](https://github.com/brigadehub/brigadehub/commit/e7f12a7acfbea045bf0690929d2631f22dcca4c7)] - **blog:**  updating controllers to handle new blog format (Trent Oswald)
- [[`0d271d74c5`](https://github.com/brigadehub/brigadehub/commit/0d271d74c5f902247daa31e60ff14a433b502de2)] - **deps:**  updated public theme to next version (Trent Oswald)
- [[`6d425976e8`](https://github.com/brigadehub/brigadehub/commit/6d425976e89779b3e0232f91546f7653162d87ef)] - **seed:**  removed defunct seed file (Trent Oswald)
- [[`81618e3edd`](https://github.com/brigadehub/brigadehub/commit/81618e3edddbee9de5fe7231af32e043e2708c2d)] - **version:**  bumped to 1.0.0-alpha.17 (Trent Oswald)


# v1.0.0-alpha.16 - Pervasive Octopus | published *2016-12-06T22:00:25Z*

## 2016-12-06, Version v1.0.0-alpha.16
### Notable Changes

- Split out themes into [brigadehub-public-c4sf](https://github.com/brigadehub/theme-public-c4sf) and [brigadehub-admin-c4sf](https://github.com/brigadehub/theme-admin-c4sf)

### Commits
- [[`0c50840aff`](https://github.com/brigadehub/brigadehub/commit/0c50840affb97dfb454d08fe8bb965cc9406c89f)] - **theme:**  split out theme to separate repo (Trent Oswald)
- [[`25dae55e09`](https://github.com/brigadehub/brigadehub/commit/25dae55e098fe23a26995913a1e07ace273d136d)] - **seed:**  changed theme slug in db seed (Trent Oswald)
- [[`2dacdb23bf`](https://github.com/brigadehub/brigadehub/commit/2dacdb23bf106118935479d5238d1dc194cf8dfb)] - **deps:**  added brigadehub-public-c4sf theme dependency (Trent Oswald)
- [[`4bdaa97571`](https://github.com/brigadehub/brigadehub/commit/4bdaa975715c5cbd6a05530ca36359d5c8f4b233)] - **db:**  added migration for deprecated theme (Trent Oswald)
- [[`a87e7f0df9`](https://github.com/brigadehub/brigadehub/commit/a87e7f0df99a421744ac5259ba52cad247ca6b39)] - **scripts:**  added migrate to postinstall (Trent Oswald)
- [[`6fc61876e4`](https://github.com/brigadehub/brigadehub/commit/6fc61876e420c5192771d91585ad41a07860a8f6)] - **lint:**  standard compliance (Trent Oswald)
- [[`adf77ae4e3`](https://github.com/brigadehub/brigadehub/commit/adf77ae4e3e27dd4d11635a7227f2a842ef8f7dc)] - **controllers:**  moved users, dashboard, checkin, home to theme (Trent Oswald)
- [[`fe45407d15`](https://github.com/brigadehub/brigadehub/commit/fe45407d1563ce772c9ad98b9002f8c7627dd691)] - **models:**  attached models/config/helpers to req (Trent Oswald)

seeds/Posts.js
var faker = require('faker')

function generateWisdom () {
  var start = ['Never forget to', 'To be human is to', 'In the end we all need to', 'Above all you will', 'Some souls will', 'Remember to', 'You deserve to', 'We were born to', 'Starting today', 'The truth is everyone will', 'Mistakes are how you learn to', 'Success is learning to', 'Take time every day to', 'Life is about remembering to']
  var verb = ['breath', 'live', 'watch', 'achieve', 'ache', 'accept', 'annunciate', 'befriend', 'blush', 'concede', 'encode', 'foretell', 'imagine', 'imply', 'inspire', 'observe', 'permit', 'ponder', 'postulate', 'rave', 'rhyme', 'snack', 'uphold', 'wander', 'worship', 'work', 'weep', 'view', 'triumph', 'travel', 'thrive', 'terrify', 'tend', 'taste', 'survive', 'suffer', 'subscribe', 'stride', 'steep', 'sneeze', 'snap', 'slink', 'shout', 'shiver', 'select', 'scrawl', 'sacrifice', 'review', 'pretend', 'proceed', 'preach']
  var noun = ['understanding', 'wisdom', 'strength', 'courage', 'love', 'silence', 'happiness', 'warmth', 'freedom', 'absurdness', 'artfulness', 'bitterness', 'candidness', 'conciseness', 'sophistication', 'backbone', 'brilliance', 'brutality', 'charity', 'coldness', 'Compassion', 'Confidence', 'Contentment', 'Courage', 'Curiosity', 'Dedication', 'Determination', 'Elegance', 'Enthusiasm', 'Envy', 'Fear', 'Generosity', 'Goodness', 'Graciousness', 'Hatred', 'Honesty', 'Honor', 'Hope', 'Humility', 'Humor', 'Insanity', 'Integrity', 'Intelligence', 'Jealousy', 'Kindness', 'Loyalty', 'Maturity', 'Patience', 'Perseverance', 'Sanity', 'Self-control', 'Sensitivity', 'Sophistication', 'Stupidity', 'Sympathy', 'Talent', 'Tolerance', 'Trust', 'Warmth', 'Weakness', 'Wisdom', 'Wit']
  var wisdom = start[Math.floor(Math.random() * start.length)] + ' ' + verb[Math.floor(Math.random() * verb.length)] + ' with ' + noun[Math.floor(Math.random() * noun.length)].toLowerCase()
  return wisdom
}

var testPosts = []

for (var i = 0; i < 30; i++) {
  var date = faker.date.past()
  var wisdom = generateWisdom()
  testPosts.push({
    'slug': wisdom.toLowerCase().replace(/\s+/g, '-'),
    'title': wisdom + i,
    'author': ['fakeUser', 'spiffysparrow', 'happyPoster'][Math.floor(Math.random() * 3)],
    'url': '',
    'image': 'http://i.imgur.com/2lHqtJ7.png',
    'description': faker.lorem.sentence(30),
    'content': '# Example post\n\nThis is a test post\n\n- one\n- two\n\ntest | test2\n---- | -----\ntester | tester2\n\n',
    'date': new Date(date),
    'tags': [
      wisdom.split(' ').pop(),
      'wisdom'
    ],
    'published': i < 25,
    sync: {
      jekyll: '',
      wordpress: '',
      medium: ''
    }
  })
}

module.exports = testPosts
